FrontEnd/src/BookingPage.jsx

```diff
@@ -0,0 +1,93 @@
+import "./BookingPage.css";
+import { useLocation } from "react-router-dom";
+import { useState } from "react";
+import { requestAppointment } from "./UserAuthentication/FirestoreDB";
+
+function BookingForm() {
+  const location = useLocation();
+  const { post, userUID } = location.state || {};
+  const [appointmentData, setAppointmentData] = useState({
+    title: "",
+    description: "",
+    date: "",
+    time: "",
+    additionalNote: "",
+  });
+  const handleChange = (event) => {
+    const { name, value } = event.target;
+    setAppointmentData({ ...appointmentData, [name]: value });
+  };
+
+  const sendAppointmentRequest = async (
+    userUID,
+    vendorUID,
+    appointmentData
+  ) => {
+    await requestAppointment(userUID, vendorUID, appointmentData);
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    sendAppointmentRequest(userUID, post.userId, appointmentData);
+  };
+
+  if (!post) {
```

Comment:
Can we have explicit null check here?
```suggestion
  if (post == null) {
```